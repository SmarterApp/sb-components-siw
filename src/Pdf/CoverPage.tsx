import * as React from "react";

export interface CoverPageProps {
  subject: string;
  grade: string;
  dateString?: string;
}

export class CoverPage extends React.Component<CoverPageProps, {}> {
  render() {
    const today = this.props.dateString || new Date().toLocaleDateString();

    return (
      <div className="page first-page">
        <div className="first-page-title">
          Smarter Balanced<br />
          Assessment Consortium
        </div>
        <div className="first-page-subtitle">
          {this.props.subject} Practice Test Scoring Guide<br />
          <strong>{this.props.grade}</strong>
        </div>
        <div className="first-page-date">{today}</div>
      </div>
    );
  }
}
