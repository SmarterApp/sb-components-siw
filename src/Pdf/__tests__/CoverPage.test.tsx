import * as React from "react";
import * as ReactDOM from "react-dom";
import { CoverPage } from "../CoverPage";
import { shallow, mount, render } from "enzyme";

describe("First page", () => {
  it("matches snapshot", () => {
    const wrapper = shallow(
      <CoverPage subject="math" grade="grade 3" dateString="01/12/2017" />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
