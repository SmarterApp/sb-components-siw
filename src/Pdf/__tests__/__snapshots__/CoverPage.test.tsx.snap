// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`First page matches snapshot 1`] = `
<div
  className="page first-page"
>
  <div
    className="first-page-title"
  >
    Smarter Balanced
    <br />
    Assessment Consortium
  </div>
  <div
    className="first-page-subtitle"
  >
    math
     Practice Test Scoring Guide
    <br />
    <strong>
      grade 3
    </strong>
  </div>
  <div
    className="first-page-date"
  >
    01/12/2017
  </div>
</div>
`;
